api/src/middleware/authz.middleware.ts

```diff
@@ -49,40 +49,80 @@ class Auth0PayloadError extends Error {
   }
 }
 
-function findOrCreateUserFromAuth0Token(token: string): Promise<User> {
-  return axios
-    .get(`${AUTH0_DOMAIN}/userinfo`, {
-      headers: {
-        authorization: token,
-      },
-    })
-    .then(async ({ data }: { data: Auth0Response }) => {
-      const sub = data.sub
-      if (isNil(sub)) {
-        throw new Auth0PayloadError(data)
-      }
-
-      const { email, given_name: firstName, family_name: lastName } = data
-
-      const fallbackEmail = `${firstName}.${lastName}@yukon-no-email.ca`
-      const [user, created] = await User.findOrCreate({
-        where: { sub },
-        defaults: {
-          sub,
-          email: email || fallbackEmail,
-          firstName,
-          lastName,
-          roles: [User.Roles.USER],
-          status: User.Statuses.ACTIVE,
-        },
-      })
-
-      if (created) {
-        logger.info(`CREATED USER FOR ${email}: ${JSON.stringify(user.dataValues)}`)
-      }
-
-      return user
-    })
+export async function findUserFromAuth0Token(token: string): Promise<User | null> {
+  const response = await axios.get(`${AUTH0_DOMAIN}/userinfo`, {
+    headers: {
+      authorization: token,
+    },
+  });
+
+  const data = response.data;
+  const sub = data.sub;
+
+  if (isNil(sub)) {
+    throw new Auth0PayloadError(data);
+  }
+
+  const user = await User.findOne({ where: { sub } });
+  return user; // NOTE null if not found
+}
+
+export async function createUserFromAuth0Token(token: string): Promise<User | null> {
+  const user = await findUserFromAuth0Token(token);
+
+  // user with token is already in database
+  if(user !== null){
+    logger.error(`ATTEMPTING TO CREATE EXISTING USER: ${JSON.stringify(user.dataValues)}`)
+    return null; // return null since no user was created
+  }
+  
+  // user with token is not in database
+  const response = await axios.get(`${AUTH0_DOMAIN}/userinfo`, {
+    headers: {
+      authorization: token,
+    },
+  });
+
+  const data: Auth0Response = response.data;
+  const sub = data.sub;
```

Comment:
Generally I'd combine these
```suggestion
  const auth0Subject = response.data?.sub
```